src/bp-core/bp-core-rewrites.php

```diff
@@ -175,7 +175,7 @@ function bp_rewrites_get_url( $args = array() ) {
 	$bp  = buddypress();
 	$url = get_home_url( bp_get_root_blog_id() );
 
-	$r = bp_parse_args(
+	$args = bp_parse_args(
```

Comment:
I don't get the difference here... 🤔

Reply:
I understand why @sjregan made this change, he defined `$r` a bit later to be the `= $args;` and this move let us preserve all parsed function arguments  for the filter at the end. Into the function $r is edited at many times that's why.
But I agree it's a bit confusing this way. I believe we should put the parsed arguments in a new variable and also send unparsed arguments into the filter. So this line should stay the way it is:
```suggestion
	$r = bp_parse_args(
```

Reply:
Got it! I figured I was missing the bigger picture here. :)